dependency_reporter/scripts/app.py

```diff
@@ -0,0 +1,68 @@
+from get_dependencies import RepoDependencies
+import os, copy, json, itertools
+
+
+def recursive(dep_dict, tuple, seen=[]):
+    circles = []
+    if tuple[0] in seen:
+        circles.append(seen[seen.index(tuple[0]) :])
+        seen = []
+    else:
+        seen.append(tuple[0])
+        if tuple[1]:
+            for item in tuple[1]:
+                circles.extend(
+                    recursive(dep_dict, [item, dep_dict.get(item)], seen.copy())
+                )
+    return circles
+
+
+def do_logic(dep_dict):
+    circles = []
+    for key in dep_dict:
+        circles.extend(recursive(dep_dict, [key, dep_dict[key]], []))
+    circles.sort()
+    with open("reports/output.json", "w") as output:
+        json.dump(list(k for k, _ in itertools.groupby(circles)), output)
+
+
+def main():
+    enterprise_url = os.getenv("ENTERPRISE_URL")
+    os_repos = os.getenv("OPENSOURCE_REPO_LIST")
+    enterprise_repos = os.getenv("ENTERPRISE_REPO_LIST")
+    os_org = os.getenv("GITHUB_ORG")
+    enterprise_org = os.getenv("ENTERPRISE_ORG")
+
+    if os_repos:
+        if not os_org:
+            raise EnvironmentError("No github.com org name provided")
+        os_repos_list = os_repos.split(" ")
```

Comment:
Potentially check for os_org first, then derive github.com org name from links. Reduce the amount of preparation by the operator.

Reply:
In fact, scratch that. I understand why you've checked for os repos first, as the users mightn't have any. I would calculate the github.com org from the repo URLs though. Kick out if an org differs for a repo in the list. This reduces the set up for the operator.